### ISSUE TITLE
Misuse of `iteration` in `train_contrastive_feature.py`

### ISSUE THREAD

**Issue by wyysteelhead** [Jumpat/SegAnyGAussians#26]
I've noticed a potential issue in `train_contrastive_feature.py`. 

In line 187:
```python
scene.save_feature(iteration, target = 'contrastive_feature')
torch.save(sam_proj.state_dict(), os.path.join(scene.model_path, "point_cloud/iteration_{}/".format(iteration) + "sam_proj.pt"))
```
The `iteration` variable here seems to be incorrectly used. It appears to be `opt.iterations` as per the loop on line 110:
```python
for iteration in range(first_iter, opt.iterations + 1):
```
However, since the model is being saved to the directory "point_cloud/iteration_{}/", it seems more likely that it should be `args.iteration` rather than `opt.iterations`. 

The problem is, if `args.iteration` is not specified, its default value would be -1, which would not work for the directory path. Therefore, simply renaming `iteration` wouldn't solve the issue.

As a workaround, I replaced `iteration` with `scene.loaded_iter`:
```python
scene.save_feature(scene.loaded_iter, target = 'contrastive_feature')
torch.save(sam_proj.state_dict(), os.path.join(scene.model_path, "point_cloud/iteration_{}/".format(scene.loaded_iter) + "sam_proj.pt"))
```
This seems to have solved the issue for me. Could you please confirm if this is the correct approach?

**Comment by Jumpat** [comment]
Hello! Thanks for pointing out this!👍

I think this modification is okay. In fact as long as the ckpt can be correctly loaded, its name is not important.

**Comment by wyysteelhead** [comment]
I initially ran `train_scene.py` for 30,000 iterations. Subsequently, I ran `train_contrastive_feature.py` for 5,000 iterations. However, I noticed that the “sam_proj.pt” file was saved in the “point_cloud/iteration_5000” directory.

Considering that I had loaded the model from iteration 30,000 in `train_contrastive_feature.py` and would base further segmentation on this model, I expected the “sam_proj.pt” file to be saved in the “point_cloud/iteration_30000” directory.

I believe this inconsistency might lead to confusion and potential errors in subsequent steps.

**Comment by Jumpat** [comment]
Yeah, you're right. This makes the code confusing, and we may fix this in future update.
